src/model.jl

```diff
@@ -289,6 +504,13 @@ julia> constraint!(ocp, :state, 2:3, [ 0, 0 ], [1, 2])
 ```
 """
 function constraint!(ocp::OptimalControlModel, type::Symbol, rg::Union{Index, UnitRange{<:Integer}}, lb, ub, label::Symbol=__constraint_label())
```

Comment:
@ocots nice to have added sanity checks in the functional routines. they are called by the abstract formulation that only performs syntax check; it is good that semantic tests are done in the functional part.